cdap-ui/app/cdap/components/AbstractWidget/SchemaEditor/Context/SchemaParser.ts

```diff
@@ -412,7 +412,10 @@ function parseSubTree(field: IFieldType | IFieldTypeNullable): INode {
  */
 function parseSchema(avroSchema: ISchemaType): INode {
   const name = avroSchema.name || 'etlSchemaBody';
-  const fields = avroSchema.schema.fields;
+  let fields = avroSchema.schema.fields;
```

Comment:
maybe we can try something like `const fields = (avroSchema || getDefaultEmptyAvroSchema()).schema.fields;`, if this does not work we can also try ternary.

Reply:
Checking for fields seemed more specific and easier to read. But I see the concern of using dot operator and have used `objectQuery` to make sure it is safe.